### DISCUSSION TITLE
Concourse  to managed test  on Hardware stuff

### DISCUSSION THREAD

**Discussion by DrummyFloyd** [concourse/concourse#8489] [General]
hi,
i'm looking to any idea or solution, for some hardware testing with Concourse.
like 
a web node in cloud cluster for build feature.
then have tagged workers on NUC or RP4, which will control some hardwares.
like this i will be able to run some test on hardware( flash/boot/test using serial port or ssh)

any idea if it's possible to achieve something like this with concourse ? 
i don't want to go on Jenkins...
regards
drummy

**Comment by marco-m** [comment]
Hello,
What have you tried? If you use the tagged worker as you mention, what is not working?

**Comment by Scottapotamas** [comment]
It's probably not the best way to do it, but I did [a writeup/blogpost of how we test hardware with Concourse workers](https://electricui.com/blog/concourse-hardware-linux) for macOS, Windows and Linux.

To summarise (in case of link-rot etc):

- There's some discussion in #7529 re: mapping devices and cgroups which you should try first if it fits your needs. I'm not running K8 so passthrough behaviour is a touch different, and that approach didn't handle device hotplug/changes which are part of the test suite.
- macOS and Windows workers don't have containerisation and therefore hardware interaction 'just works',
- therefore, I setup a specific Linux worker using the `houdini` runtime, this makes it behave like macOS/Win.
- The downside of this approach is the loss of containerisation for these steps, and therefore package/dep management and security considerations. We just try to do as much as possible with our normal Linux worker before touching hardware.

**Reply by marco-m** [reply]
@Scottapotamas thanks for sharing your workflow. It is the same I would do for this use case.
